python-package/lightgbm/plotting.py

```diff
@@ -10,24 +11,36 @@
 from .sklearn import LGBMModel
 
 
-def _check_not_tuple_of_2_elements(obj, obj_name='obj'):
+def _check_not_tuple_of_2_elements(obj: Any, obj_name: str = 'obj') -> None:
     """Check object is not tuple or does not have 2 elements."""
     if not isinstance(obj, tuple) or len(obj) != 2:
         raise TypeError(f"{obj_name} must be a tuple of 2 elements.")
 
 
-def _float2str(value, precision=None):
+def _float2str(value: float, precision: Optional[int] = None) -> str:
     return (f"{value:.{precision}f}"
             if precision is not None and not isinstance(value, str)
             else str(value))
 
 
-def plot_importance(booster, ax=None, height=0.2,
-                    xlim=None, ylim=None, title='Feature importance',
-                    xlabel='Feature importance', ylabel='Features',
-                    importance_type='split', max_num_features=None,
-                    ignore_zero=True, figsize=None, dpi=None, grid=True,
-                    precision=3, **kwargs):
+def plot_importance(
+    booster: Union[Booster, LGBMModel],
+    ax=None,
+    height: float = 0.2,
+    xlim: Optional[Tuple[int, int]] = None,
+    ylim: Optional[Tuple[int, int]] = None,
+    title: str = 'Feature importance',
+    xlabel: str = 'Feature importance',
+    ylabel: str = 'Features',
+    importance_type: str = 'split',
+    max_num_features: Optional[int] = None,
+    ignore_zero: bool = True,
+    figsize: Optional[Tuple[int, int]] = None,
```

Comment:
`figsize` is tuple of floats internally, not ints.

> figsize(float, float), default: rcParams["figure.figsize"] (default: [6.4, 4.8])
https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.figure.html